src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py

```diff
@@ -250,6 +250,111 @@ def from_pretrained(cls, *args, **kwargs):
+            encoder(encoder.dummy_inputs)
+            decoder(decoder.dummy_inputs)
+
+            # Get the variable correspondence between `_tf_model` and `encoder` and `decoder`
+            encoder_variables = {}
+            for variables in [encoder.trainable_variables, encoder.non_trainable_variables]:
+                for v in variables:
+                    encoder_variables["/".join(v.name.split("/")[1:])] = v
+            decoder_variables = {}
+            for variables in [decoder.trainable_variables, decoder.non_trainable_variables]:
+                for v in variables:
+                    decoder_variables["/".join(v.name.split("/")[1:])] = v
+
+            _encoder_variables = {}
+            for variables in [_tf_model.encoder.trainable_variables, _tf_model.encoder.non_trainable_variables]:
+                for v in variables:
+                    _encoder_variables["/".join(v.name.split("/")[2:])] = v
+            _decoder_variables = {}
+            for variables in [_tf_model.decoder.trainable_variables, _tf_model.decoder.non_trainable_variables]:
+                for v in variables:
+                    _decoder_variables["/".join(v.name.split("/")[2:])] = v
+
+            # assign weight values to `encoder` and `decoder` from `_tf_model`
+            for name, v in encoder_variables.items():
+                v.assign(_encoder_variables[name])
+            for name, v in decoder_variables.items():
+                v.assign(_decoder_variables[name])
+
+            tf_model = TFVisionEncoderDecoderModel(encoder=encoder, decoder=decoder)
+            tf_model.config = _tf_model.config
+
+            # Deal with `enc_to_dec_proj`
+            if hasattr(_tf_model, "enc_to_dec_proj"):
+                tf_model(tf_model.dummy_inputs)
+                tf_model.enc_to_dec_proj.kernel.assign(_tf_model.enc_to_dec_proj.kernel)
+                tf_model.enc_to_dec_proj.bias.assign(_tf_model.enc_to_dec_proj.bias)
+
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                encoder_dir = os.path.join(tmpdirname, "encoder")
+                decoder_dir = os.path.join(tmpdirname, "decoder")
+                tf_model.encoder.save_pretrained(encoder_dir)
+                tf_model.decoder.save_pretrained(decoder_dir)
+                model = VisionEncoderDecoderModel.from_encoder_decoder_pretrained(
+                    encoder_dir, decoder_dir, encoder_from_tf=True, decoder_from_tf=True
+                )
+                # This is only for copying some specific attributes of this particular model.
+                model.config = tf_model.config
+
+                if hasattr(tf_model, "enc_to_dec_proj"):
+                    model.enc_to_dec_proj.weight.data = torch.transpose(
+                        torch.from_numpy(tf_model.enc_to_dec_proj.kernel.numpy()), 1, 0
+                    )
+                    model.enc_to_dec_proj.bias.data = torch.from_numpy(tf_model.enc_to_dec_proj.bias.numpy())
```

Comment:
`from_encoder_decoder_pretrained` can't deal with `enc_to_dec_proj`, so we need to work on it here